packages/odyssey-react/src/components/Table/Table.test.tsx

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import { render } from "@testing-library/react";
+import Table from ".";
+import TableHeader from "./TableHeader";
+import TableBody from "./TableBody";
+import TableFooter from "./TableFooter";
+
+const caption = 'test table'
+
+describe("Table", () => {
+  it("renders the table", () => {
+    const { getByRole } = render(
+      <Table caption={caption}>
+        <TableHeader />
+        <TableBody />
+        <TableFooter />
+      </Table>
+    );
+
+    expect(getByRole('table')).toBeInTheDocument();
+  });
+
+  it('renders the caption prop in the caption element', () => {
+    const { getByRole } = render(
+      <Table caption={caption} />
+    );
+
+    expect(getByRole('table', { name: caption })).toBeTruthy();
+  });
+
+  it('adds a custom class', () => {
```

Comment:
I understand the logic, but I'd be very hesitant to lock down the api in that way at this early stage. In my experience it only leads to worse behavior on the part of developers (styling with id, overrides with classes on the container, fishing in to the DOM directly).

Especially in the case of Table where we provide no functionality and developers will need maximum flexibility.

Reply:
@andrewberg-okta @arnoldsandoval-okta @edburyenegren-okta lets discuss more in person. For this PR I don't think we should diverge from the approach being used for other components unless we have a discussion and plan to update everything.

Reply:
And it's not just styles to consider, there may be cases of downstream testing and tooling needing access to the className attribute.

Reply:
@andrewberg-okta good thoughts - I have worked on products and teams that followed this approach. My hope with this restriction is that we can make maintaining the system across upgrades easier, enforce better design consistency, and help teams write better tests. It will never be perfect, if we're spreading omitted props a clever developer could apply a `data-foo` attribute today (which are not omitted) and use a corresponding selector to target it the same way they would a normal. I recognize no preventative measures are perfect but I primarily want to minimize the effort and pain we'll be on the hook for if folks get into the habit of expecting their cascade based overrides to work against our system in every case across every version.

The long term alternative is of course having a robust, mature, theming system and set of variants to provide style customizations. I detail that vision in [this RFC here](https://docs.google.com/document/d/1Lg1YXMTtr6H5IEmf6ptQTO_DMoAq2dOdttzYcxiblZc/edit) and with our CSS Modules Refactor epic we're making the first step toward that goal!